pheps/phep-0003.md

```diff
@@ -0,0 +1,374 @@
+            str(rel_min.strftime("%b %Y"))
+            if rel_min == rel_max
+            else f"{rel_min.strftime('%b %Y')} and {rel_max.strftime('%b %Y')}"
+        )
+        table.append(f"|{package:<15}|{version_range:<19}|released {rel_range}|")
+
+    return pad_table(table)
+
+
+def make_adopt_table(sub):
+    table = []
+    table.append("|    |    |    |")
+    table.append("|----|----|----|")
+    for package in sorted(set(sub.index.get_level_values(0))):
+        vers = sub.loc[[package]]["version"]
+        minv, maxv = min(vers), max(vers)
+        support_bys = sub.loc[[package]]["support_by"]
+        support_by_min, support_by_max = min(support_bys), max(support_bys)
+        version_range = str(minv) if minv == maxv else f"{minv} to {maxv}"
+        support_by_range = (
+            str(support_by_min.strftime("%b %Y"))
+            if support_by_min == support_by_max
+            else f"{support_by_min.strftime('%b %Y')} and {support_by_max.strftime('%b %Y')}"
+        )
+        table.append(f"|{package:<15}|{version_range:<19}|support by {support_by_range}|")
+
+    return pad_table(table)
+
+
+def make_quarter(quarter, dq_drop, dq_support_by):
+    table = ["#### " + str(quarter).replace("Q", " - Quarter ") + ":\n"]
+    
+    # Add new versions adoption schedule if not empty
+    if quarter in dq_support_by.index.get_level_values(0):
+        table.append("###### Adopt support for:\n")
+        adopt_sub = dq_support_by.loc[quarter]
+        adopt_table = make_adopt_table(adopt_sub)
+        table.extend(adopt_table)
+
+    table.append("\n###### Can drop support for:\n")
+    sub = dq_drop.loc[quarter]
+    table.extend(make_table(sub))
+    
+    return "\n".join(table)
+
+
+with open("schedule.md", "w") as fh:
+    # we collect package 6 month in the past, and drop the first quarter
+    # as we might have filtered some of the packages out depending on
+    # when we ran the script.
+    tb = []
+    for quarter in list(sorted(set(dq_drop.index.get_level_values(0))))[1:]:
+        tb.append(make_quarter(quarter, dq_drop, dq_support_by))
+
+    fh.write("\n\n".join(tb))
+    fh.write("\n")
+
+```
+
+# Rejected Ideas
+<a name="rejected-ideas"></a>
+ - [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html)'s more lenient 42-month support timeline was originally considered instead of [SPEC 0](https://scientific-python.org/specs/spec-0000/)'s 36 months, but it was ultimately decided to follow SPEC 0 because it supersedes NEP 29.
+ - The scope of this PHEP was originally limited to Python version support. 
+ However, it was decided that including the upstream package support policy from SPEC 0 would better promote PyHC package interoperability and avoid the need for a future separate PHEP.
+
+# Open Issues
+<a name="open-issues"></a>
+1. What should go in the "How to Teach This" section? Should we expand on the ideas already there or take it a different direction?
+Or leave it if it is sufficient already?
```

Comment:
Note that in the absence of feedback I'd now consider "leave it" the default option here.